### ISSUE TITLE
Add support to the `Prefer` and `Preference-Applied` HTTP headers defined in the RFC 7240

### ISSUE THREAD

**Issue by cassiomolin** [spring-projects/spring-framework#22360]
Looks like the [`HttpHeaders`][1] class doesn't support the [`Prefer`][3] and [`Preference-Applied`][4] HTTP headers defined in the [RFC 7240][2].

Once those headers are [registered in IANA][5], it would be great to have support for them.

  [1]: https://docs.spring.io/spring-framework/docs/5.1.4.RELEASE/javadoc-api/org/springframework/http/HttpHeaders.html
  [2]: https://tools.ietf.org/html/rfc7240
  [3]: https://tools.ietf.org/html/rfc7240#section-2
  [4]: https://tools.ietf.org/html/rfc7240#section-3
  [5]: https://www.iana.org/assignments/message-headers

**Comment by bclozel** [comment]
Looking at the RFC, it seems that the only thing strictly defined is the header format.
Other than that, it's pretty much up to the server and clients to agree beforehand about the semantics of header value.

>   Similarly, servers that process requests are often faced with
>   decisions about how to process requests that may be technically
>   invalid or incorrect but are still understandable.  It might be the
>   case that the server is able to overlook the technical errors in the
>   request but still successfully process the request.  Depending on the
>   specific requirements of the application and the nature of the
>   request being made, the client might or might not consider such
>   lenient processing of its request to be appropriate.

I don't see how Spring Framework could leverage those headers in the existing MVC / WebFlux programming model. Is there some specific piece of information that I missed in the RFC? Are other frameworks using it for some specific behavior?

What is the feature you are actually requesting?

**Comment by cassiomolin** [comment]
> Are other frameworks using it for some specific behavior?

I'm not aware of any other frameworks that currently relies on those headers for some specific behavior.

> What is the feature you are actually requesting?

I think the original description is a bit poor, so let me rephrase it and clarify what I meant with _"support"_.

The [`HttpHeaders`][1] class currently doesn't have any constants for the [`Prefer`][3] and the [`Preference-Applied`][4] HTTP headers defined in the [RFC 7240][2].

I've been working on a few projects that required such headers and I found myself defining my own constants for them. Once those headers are [registered in IANA][5], it would be handy to have constants for them.


  [1]: https://github.com/spring-projects/spring-framework/blob/5.1.x/spring-web/src/main/java/org/springframework/http/HttpHeaders.java
  [2]: https://tools.ietf.org/html/rfc7240
  [3]: https://tools.ietf.org/html/rfc7240#section-2
  [4]: https://tools.ietf.org/html/rfc7240#section-3
  [5]: https://www.iana.org/assignments/message-headers
